Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -85,37 +79,34 @@ final class ConnectionPool {
     }
 
     func release(_ connection: Connection) {
-        let connectionProvider = self.connectionProvidersLock.withLock {
-            self.connectionProviders[connection.key]
+        let connectionProvider = self.lock.withLock {
```

Comment:
Almost certainly, this is racy. I’m pretty sure release could be called twice (eg pool shutdown vs ordinary connection release). Both could then call release with the same connection and that probably leads to bad things. Am I missing something?

Reply:
I will test this

Reply:
this will not be called twice, since we are not calling it on pool shutdown, on pool shutdown we call close on connections and tasks, and they as part of their shutdown call release, does that make sense?

Reply:
👍